XenAdmin/Wizards/NewNetworkWizard.cs

```diff
@@ -202,10 +203,27 @@ private void CreateCHIN()
             (new CreateChinAction(xenConnection, network, theInterface)).RunAsync();
         }
 
+
+        private DialogResult ShowSriovCreationWarning()
+        {
+            var dlg = new ThreeButtonDialog(
+                new ThreeButtonDialog.Details(
+                    SystemIcons.Warning, 
+                    Messages.SRIOV_NETWORK_CREATE_WILL_DISTURB_CONNECTION),
+                 new ThreeButtonDialog.TBDButton(Messages.SRIOV_NETWORK_CREATE, DialogResult.OK),
+                ThreeButtonDialog.ButtonCancel);
+
+            dlg.ShowDialog(this);
+            return dlg.DialogResult;
+        }
+
         private void CreateSRIOV()
         {
             XenAPI.Network network = PopulateNewNetworkObj();
 
+            if (ShowSriovCreationWarning() != DialogResult.OK)

```

Comment:
This should be the first call in this function, before the line ```XenAPI.Network network = PopulateNewNetworkObj();```. Actually, because there are several `return`s in this function, the network variable should be moved closer to the place where it is used.